src/components/Home.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { render } from "@testing-library/react";
+import { Provider } from "react-redux";
+import store from "../store";
+import Home from "./Home";
+import { authUser } from '../store/userSlice';
+
+describe("Home component", () => {
+  test("renders 'Please log in' message when user is null", () => {
+    const { getByText } = render(
+      <Provider store={store}>
+        <Home />
+      </Provider>
+    );
+
+    expect(getByText("Please log in")).toBeInTheDocument();
+  });
+
+  test("renders 'Welcome, {username}!' message when user is not null", () => {
+    const username = "JohnDoe";
+    store.dispatch(authUser({ username }));
```

Comment:
Dispatching to the real/main store means that if you add another test, it will still use the dirty store with the data from the previous test. You might want to create some kind of a mock store instead (for example, it can be `mockStore(initialState)` and you can fill up the initial state with whatever the test needs) and generate/construct/initialise it within each test